### ISSUE TITLE
"key" not found in archive.

### ISSUE THREAD

**Issue by jaf91** [YuvrajRaghuvanshiS/WhatsApp-Key-Database-Extractor#73]
**Describe the bug**
After successfully unpacked tmp/whatsapp.ab, I get the error: bin\tar.exe: apps/com.whatsapp/f/key: Not found in archive 
**To Reproduce**
1. run 'py wa_kdbe.py --allow-reboot'
2. See error

**Terminal output**
```

[15:25:32.135413] Current release date : 29/06/2021


[15:25:32.137406] Please read above instructions carefully ↑ . Continue? (default y) : y


[15:25:45.590360] If you haven't already, it is adviced to take a WhatsApp chat backup by going to WhatsApp settings → Chat Settings → Chat Backup. Hit Enter key to continue.
[15:25:47.385492] Connected to SM-J710MN
[15:25:51.622885] WhatsApp V2.21.15.20 installed on device
[15:25:51.624659] Found legacy WhatsApp V2.11.431 apk in helpers/ folder
[15:25:52.916548] Backing up WhatsApp 2.21.15.20 apk, the one installed on device to tmp/WhatsAppbackup.apk
/sdcard/WhatsAppbackup.apk: 1 file pulled, 0 skipped. 22.4 MB/s (35278159 bytes in 1.502s)
[15:25:55.255573] Apk backup complete.
[15:25:55.256522] Uninstalling WhatsApp, skipping data.
Success
[15:25:57.116690] Uninstalled.


[15:25:57.122670] Rebooting device, please wait.
[15:25:59.764903] Waiting for device...
[15:26:05.198433] Waiting for device...
[15:26:10.296512] Waiting for device...
[15:26:15.429150] Waiting for device...
[15:26:20.535821] Waiting for device...
[15:26:25.903811] Waiting for device...
[15:26:30.985005] Waiting for device...
[15:26:36.066944] Waiting for device...
[15:26:41.270304] Waiting for device...
[15:26:46.344071] Hit Enter key after unlocking device.
[15:26:48.439956] Installing legacy WhatsApp V2.11.431, hold tight now.
Performing Streamed Install
Success
[15:27:16.583067] Installation Complete.
Starting: Intent { cmp=com.whatsapp/.Main launchParam=MultiScreenLaunchParams { mDisplayId=0 mBaseDisplayId=0 mFlags=0 } }
[15:27:19.841462] Hit Enter key after running Legacy WhatsApp for a while. Ignore invalid date warning.
[15:27:41.079705] Backing up WhatsApp data as tmp/whatsapp.ab. May take time, don't panic.
WARNING: adb backup is deprecated and may be removed in a future release
Now unlock your device and confirm the backup operation...
[15:27:54.850835] Done backing up data. Size : 5325360 bytes.
[15:27:54.851833] Reinstallting original WhatsApp.
Performing Streamed Install
Success


[15:28:24.622327] Our work with device has finished, it is safe to remove it now.


[15:28:24.625315] Found whatsapp.ab in tmp folder. Continuing... Size : 5325360 bytes.
[15:28:24.627310] Enter a name for this user (default "user"). :
[15:28:40.790405] Enter same password which you entered on device when prompted earlier. :
[15:28:44.494607] Successfully unpacked tmp/whatsapp.ab to tmp/whatsapp.tar. Size : 16258560 bytes.
[15:28:44.495607] Folder extracted/ already exists.
[15:28:44.496602] Folder extracted/user already exists.
[15:28:44.496602] Taking out main files in tmp/ folder temporaily.
bin\tar.exe: apps/com.whatsapp/f/key: Not found in archive
bin\tar.exe: Error exit delayed from previous errors
[15:28:44.582810] [WinError 3] El sistema no puede encontrar la ruta especificada: 'tmp/apps/com.whatsapp/f/key' -> 'extracted/user/key'
[15:28:44.582810] Cleaning up tmp folder...


[15:28:44.592172] Exiting...
[15:28:44.701973] Hit 'Enter' key to continue....y


```

**Desktop (please complete the following information):**
 - OS: Windows 10
**Smartphone (please complete the following information):**
 - Device: Samsung Galaxy J7
 - Android 7
 - Samsung Experience 8.1

**Comment by YuvrajRaghuvanshiS** [comment]
This is weird. Try running it with `--tar-only` flag that way this script won't try to extract and pull important files and just leave you with tar file. There you should find key and other files.

**Comment by jaf91** [comment]
Hi! thank you very much for your help and time. I used both methods unsuccesfully.

1) with --tar-only
It manages to extract the tar file and I unzipped it, but aparently theres no key file anywhere.

![image](https://user-images.githubusercontent.com/87679563/129991382-6b874940-b786-40d3-9941-a729c23de0b1.png)

2) python's inbuilt tarfile.
I get these errors:


[18:57:15.764995] Found whatsapp.ab in tmp folder. Continuing... Size : 5239983 bytes.
[18:57:15.765994] Enter a name for this user (default "user"). :
[18:57:44.023674] Enter same password which you entered on device when prompted earlier. :
[18:57:47.405485] Successfully unpacked tmp/whatsapp.ab to tmp/whatsapp.tar. Size : 15712768 bytes.
[18:57:47.406480] Folder extracted/ already exists.
[18:57:47.409505] Folder extracted/user already exists.
[18:57:47.410468] Taking out main files in tmp/ folder temporaily.
[18:57:47.442477] key is not present in tarfile, Go and write @ 'https://github.com/YuvrajRaghuvanshiS/WhatsApp-Key-Database-Extractor/issues/73'
[18:57:47.444105] msgstore.db is not present in tarfile, Go and write @ 'https://github.com/YuvrajRaghuvanshiS/WhatsApp-Key-Database-Extractor/issues/73'
[18:57:47.450092] Extracted successfully in tmp/ : wa.db
[18:57:47.451089] Copied to extracted/user successfully : wa.db
[18:57:47.455080] Extracted successfully in tmp/ : axolotl.db
[18:57:47.456077] Copied to extracted/user successfully : axolotl.db
[18:57:47.457074] chatsettings.db is not present in tarfile, Go and write @ 'https://github.com/YuvrajRaghuvanshiS/WhatsApp-Key-Database-Extractor/issues/73'
[18:57:47.459072] Cleaning up tmp folder...
[18:57:47.465086] You should not leave these extracted database and other files hanging in folder, it is very insecure.
[18:57:47.465973] Would you like to create a password protected archive? (default y) : n

**Comment by YuvrajRaghuvanshiS** [comment]
That is not right. While using tar-only had you gone through all folder in that tar? Usually these files are there.

**Comment by YuvrajRaghuvanshiS** [comment]
The thing is I no longer can test this repo as I have deleted my whatsapp long ago otherwise it would have been a lot easier

**Comment by YuvrajRaghuvanshiS** [comment]
Copy this code 

```python
import tarfile

tar = tarfile.open('whatsapp.tar')
names = tar.getnames()
filesToSearch = ['key', 'msgstore.db',
                 'wa.db', 'axolotl.db', 'chatsettings.db']

for file in filesToSearch:
    for name in names:
        if(file in name):
            print(name)
tar.close()

input('\n\nPaused... Copy terminal now!')

```
and put in same directory where whatsapp.tar is saved. Create a flie named `search.py` paste this code in that file and run it.
Send me output.


![image](https://user-images.githubusercontent.com/52948878/130027447-bc82859b-6f58-491a-9192-acd81780c027.png)

**Comment by Lavish883** [comment]
> Copy this code
> 
> ```python
> import tarfile
> 
> tar = tarfile.open('whatsapp.tar')
> names = tar.getnames()
> filesToSearch = ['key', 'msgstore.db',
>                  'wa.db', 'axolotl.db', 'chatsettings.db']
> 
> for file in filesToSearch:
>     for name in names:
>         if(file in name):
>             print(name)
> tar.close()
> 
> input('\n\nPaused... Copy terminal now!')
> ```
> 
> and put in same directory where whatsapp.tar is saved. Create a flie named `search.py` paste this code in that file and run it.
> Send me output.
> 
> ![image](https://user-images.githubusercontent.com/52948878/130027447-bc82859b-6f58-491a-9192-acd81780c027.png)

 I got this in terminal


apps/com.whatsapp/sp/keystore.xml
apps/com.whatsapp/db/msgstore.db-shm
apps/com.whatsapp/db/msgstore.db-wal
apps/com.whatsapp/db/msgstore.db
apps/com.whatsapp/db/wa.db-shm
apps/com.whatsapp/db/wa.db-wal
apps/com.whatsapp/db/wa.db
apps/com.whatsapp/db/axolotl.db-shm
apps/com.whatsapp/db/axolotl.db-wal
apps/com.whatsapp/db/axolotl.db
apps/com.whatsapp/db/chatsettings.db

**Comment by YuvrajRaghuvanshiS** [comment]
I don't see any "path changes" of files. All are present where they need to. Talking of 'key' I can't say why it is not in output in your case. 
If key was to be in .ab file that has to show here in output. 
I am keeping this open of a month or so to see if others have same issue.

**Comment by Lavish883** [comment]
Ok thx for the info.

**Comment by YuvrajRaghuvanshiS** [comment]
Closing because of inactivity.

**Comment by tobilektri** [comment]
I have the same problem as both of them. But also i dont have success with  `--tar-only`

For my understanding: The key should be auto-generated after installing the old WA-Version, correct?
-> How does this work? 
- Is there no need for the Phone-number-activation?
- Is there no need that Whatsapp really opens the Backup?

Else: Do you have another Idea how to open my msgstore.db.crypt12 ? 
Your mentioned https://github.com/Tkd-Alex/WhatsDump/ does not work on my side, but because it has problem with the SDK

And there is also this Repo: https://github.com/EliteAndroidApps/WhatsApp-Key-Generator
But I am not able to find out what he uses as account seed while he writes
`Please do not raise any issues concerning what is used as the account seed. Figuratively speaking; you decide!`

Thanks!

**Comment by YuvrajRaghuvanshiS** [comment]
Old version is installed just to get backup permission. Nothing new is added or removed internally as per my understanding. Whatever is in there is pulled by this script. I really don't have any idea of absence of key. Maybe WA is making that key cloud based and is fetching from servers only when creating backup.

That key generator tool you mentioned "demonstrates" how WA generates the key. Seed on the other hand is unknown to anyone which makes is hard to regenerate/copy.

Stay healthy,
Yuvraj Raghuvanshi.

-------- Original Message --------

**Comment by 1c3t3a** [comment]
I've got the same problem. All files are present, only the key is missing. Did you found any solutions @tobilektri, @Lavish883?

**Comment by tobilektri** [comment]
@1c3t3a 
> I've got the same problem. All files are present, only the key is missing. Did you found any solutions?

Nope, I'm sorry. I only have a opinion what the problem was in my case. 
I did an Android update and after that Whatsapp was not able to read it's own directory. Because of encrypted backups i was not able to restore them after a reinstallation. 

Before reinstallation:
`/storage/emulated/0/WhatsApp`
After Reinstallation:
`/storage/emulated/0/Android/media/com.whatsapp/WhatsApp`

so in my case I'm sure that the problem was something with the access rights.

**Comment by simonhough** [comment]
WhatsApp is not a secure service. I was the victim of some kind of hack on Whatsapp and my account was subsequently  banned. I still do not know to this day what happened. I was in the process of blocking and reporting spammers when I got  logged off of Whatsapp and have been unable to get back in. When I email Whatsapp I just get automated responses .

Anyway I thought it would be worth trying to obtain the encryption key as I wanted to decrypt the messages that are on my phone. I used the application WhatsApp-Key-DB-Extractor-master to try and obtain my encryption keys. However the only files I get back are axolotl and wa. 

Would I be correct in saying that WhatsApp deletes your encryption key if you get banned from their service, or is it worth my time rooting my phone to get to the encryption key?

**Comment by YuvrajRaghuvanshiS** [comment]
I thought it was some device specific issue but now I have a gut feeling that WhatsApp is moving database (and other files) to some different location. In Android /data/data/com.whatsapp is safest directory as per my knowledge. If WhatsApp is moving on from this then I believe they are moving these critical files to their own cloud somewhere, since GDrive is also not that safe to store key.

If that's the case then this entire project will be nothing more than error throwing python. Poor me, this repo is not even in GitHub's Arctic archive program and I'm implementing a GUI for it.

I am not sure about the banning situation but that is possible that they delete key and other data.
I would advice against rooting your phone because it will erase everything in that process => No WhatsApp to pull data from. Also as I have mentioned above if key were to be there then my app would pull it. Its just that those files are no longer there for many users.

**Comment by simonhough** [comment]
You may be correct about the key no longer being stored on the phone. Although an alternative and I think more likely explanation is that the key does stay on your phone but Whatsapp revokes or removes the key under certain circumstances. Most of the users on the following forum still appear to be able to gain access to the key: https://forum.xda-developers.com/t/tool-whatsapp-key-db-extractor-crypt6-12-non-root-updated-october-2016.2770982/page-35 

I'm surprised Whatsapp can do this. I feel gutted that I no longer have access to my own personal data. Whatsapp should provide a means to access your information even if you are banned.

By rooting my phone would it delete the key in the process? It's an old phone which I don't care about. I'm only interested in obtaining the encryption key so I can read my messages.

**Comment by YuvrajRaghuvanshiS** [comment]
My app is based on the link you have shared.

These tech giants roll out updated to a small group then to a bigger one and since they have no limit on how much data constantly fetch from the users it is very easy for them to track what is going wrong with some feature for example.

WhatsApp shouldn't do this, I am also a strong believer that chats are the property of user and not the service providers'.

I am not sure about specific rooting process of every device but in almost all the cases rooting requires you to do a "format data" otherwise it would not even let to boot your phone. It has something to do with data encryption which is default on every Android.

What I can suggest is (though it is also risky) that if the bootloader is unlocked you install a custom recovery OS then you can access the data from within recovery. Though I am not sure if this also requires format data or not.

Stay healthy,
Yuvraj Raghuvanshi.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by simonhough** [comment]
Thanks for your reply. So you think there's still hope that I can obtain the encryption key? I did try rooting my phone but found it too complicated. I think I'll just wait until a smart person such as yourself develops an app that can successfully fetch the encryption key, assuming the key is still on the phone.

**Comment by YuvrajRaghuvanshiS** [comment]
I really have no idea what's going on with WhatsApp and until some developers on xda find it interesting enough to dig up new location of key I really can not do much as I do not have much expertise in such advanced analysis.

Yes. You can try with `--tar-only` flag to pull everything out not just those 5 files and search for key in that archive but others didn't have any luck with that also. Key is not there to begin with.

Rooting is complicated and I am sure that it will ask for data wipe, that should be last option. But if you root then you can search other directories that WhatsApp might be using other than /data/data/com.whatsapp.

The thing is that I have just re-written this project in python (originally it was in powershell, cmd, bash script). I did not invent this method, I have a general idea of why this works but I really do not have any other tricks up in my hat as a workaround.

Stay healthy,
Yuvraj Raghuvanshi.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by YuvrajRaghuvanshiS** [comment]
On newer versions of Android media files are stored in Android/media/<package_name>. So saving app specific files in Android/data/<package_name> is also an option for apps. You can access that folder with adb command (for now at least). Connect phone get into shell by typing in 'adb shell' in cmd and navigate to folder using 'cd /sdcard/Android/data/com.whatsapp' and check there. Do let me know how it goes.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by simonhough** [comment]
The old phone which I want to retrieve my WhatsApp messages is using Android 7 so there probably isn't any point using adb shell. I'm not familiar with the command prompt and I think the method you just described would involve rooting the phone and i'm not willing to do that yet as I may delete the key in the process, assuming its on the phone to begin with

**Comment by Master-TD** [comment]
It seems to me that when transferring whatsapp to another phone, the key on the old phone does not remain.

**Comment by Tanner85** [comment]
Hi guys, today I used a tool that extracted whatsapp messages. So it looks like the key is there, but with another name. It's not "key" file anymore, but obscured by name. I will let you know. The program that is able to retrieve messages is called "ULTDATA"

**Comment by ElDavoo** [comment]
What if these people had the E2E backup activated? In that case key is encrypted_backup.key

**Comment by vitormarcal** [comment]
I had the same problem that was solved when I found [this link: ](https://forum.xda-developers.com/t/tool-whatsapp-key-db-extractor-crypt6-12-non-root-updated-october-2016.2770982/page-31#post-83629013).
I guess the main step to resolve the problem is the 7 to 11.
I will reproduce  the comment bellow if the link not exists anymore: 

1. launch whatsappkeydbextract.bat file
2. input y to reboot the device
3. once rebooted, unlock it (you should not find your wa app anymore) and press any key in cw
4. wait until the streamed install is done
5. unlock again your phone
6. prompted "wa has been updated" should be prompted in your phone, press continue
7. search for wa in your phone and open it
8. (a new prompt in your phone will say it's an outdated version of wa press ok, if it doesnt appear don't worry)
9. press the adjust date and you will be sent to a date & time page, keep it open
10. press any key in cw
11. select the option you prefer at the command window
12. select fullback up in your phone (without entering any password)
13. restore wa press any key
14. you made it as well

**Comment by bogdank0** [comment]
I opened ! Who can not find the key - so you do not need it). Yes exactly ! If there is no key, then chats can simply be opened)

I did with "--tar-only" and got 4 folders ( db/f/r/sp ). And then follow these instructions: [https://github.com/andreas-mausch/whatsapp-viewer/issues/151#issuecomment-1306391834](url)  ( This is if you have an error something about the old WhatsApp Viewer)

I've been looking for a key to an open door for a week) Do not decrypt the file, just open the file com.whatsapp\db\msgstore.db

Good luck to all chat miners)

**Comment by AHM4U** [comment]
> ![IMG_20221206_013821_306](https://user-images.githubusercontent.com/119637525/205768439-866e5e77-6203-4bc9-b30b-37bed24574a2.jpg)
> 
> Can anyone help? Could not install WhatsApp... [INSTAL FAILED INVALID URI ]
> 
> install by running "restore_whatsapp.py" or manually installing from Play Store. However if it crashes then you have to clear storage/clear data from "Settings → App Settings → WhatsApp" -
> 
> 1. If I clear the data, delete it and download the application again - I will lose all the data that I want to restore.  Is that so?
> 2. but it extracts all files except the key. ??? Can't extract the key because of this crashing?  Can the key simply not be there anymore due to the block of the account?
> 
> Any guesses? I will be grateful for your help)

Hello there u can contact me on my WhatsApp I can help u out !

**Comment by Ahmnonymous** [comment]
> > ![IMG_20221206_013821_306](https://user-images.githubusercontent.com/119637525/205768439-866e5e77-6203-4bc9-b30b-37bed24574a2.jpg)
> > Can anyone help? Could not install WhatsApp... [INSTAL FAILED INVALID URI ]
> > install by running "restore_whatsapp.py" or manually installing from Play Store. However if it crashes then you have to clear storage/clear data from "Settings → App Settings → WhatsApp" -
> > 
> > 1. If I clear the data, delete it and download the application again - I will lose all the data that I want to restore.  Is that so?
> > 2. but it extracts all files except the key. ??? Can't extract the key because of this crashing?  Can the key simply not be there anymore due to the block of the account?
> > 
> > Any guesses? I will be grateful for your help)
> 
> Hello there u can contact me I can help u out but in private!

**Comment by hahn23** [comment]
> > > ![IMG_20221206_013821_306](https://user-images.githubusercontent.com/119637525/205768439-866e5e77-6203-4bc9-b30b-37bed24574a2.jpg)
> > > Can anyone help? Could not install WhatsApp... [INSTAL FAILED INVALID URI ]
> > > install by running "restore_whatsapp.py" or manually installing from Play Store. However if it crashes then you have to clear storage/clear data from "Settings → App Settings → WhatsApp" -
> > > 
> > > 1. If I clear the data, delete it and download the application again - I will lose all the data that I want to restore.  Is that so?
> > > 2. but it extracts all files except the key. ??? Can't extract the key because of this crashing?  Can the key simply not be there anymore due to the block of the account?
> > > 
> > > Any guesses? I will be grateful for your help)
> > 
> > 
> > Hello there u can contact me I can help u out but in private!

If you have the files it shows in the logs "msgstore.db" and "wa.db" than you do not need any key at all. because these files are already decrypted! You just have to open them in the whatsappviewer. If they where encrypted the would have an other file extension like msgstore.db.crypt14 or msgstore.db.crypt15.

**Comment by adi-works** [comment]
I get the same errors. I ran with the `--tar-only` argument to keep the tar file, but the script is correct, the files are indeed missing. I also noticed the script doesn't wait for the backup to end, seems like a funny racing condition. I wait a bit and then hit the "Backup" button. When it picks up on the backup correctly, it reflects in the backup size (much larger).

![image](https://github.com/YuvrajRaghuvanshiS/WhatsApp-Key-Database-Extractor/assets/12032248/4b04b215-6691-4369-85fb-3ea86863f0d0)

**Comment by hahn23** [comment]
> I get the same errors. I ran with the `--tar-only` argument to keep the tar file, but the script is correct, the files are indeed missing. I also noticed the script doesn't wait for the backup to end, seems like a funny racing condition. I wait a bit and then hit the "Backup" button. When it picks up on the backup correctly, it reflects in the backup size (much larger).
> 
> ![image](https://private-user-images.githubusercontent.com/12032248/298311337-4b04b215-6691-4369-85fb-3ea86863f0d0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU4NDQ3NjAsIm5iZiI6MTcwNTg0NDQ2MCwicGF0aCI6Ii8xMjAzMjI0OC8yOTgzMTEzMzctNGIwNGIyMTUtNjY5MS00MzY5LTg1ZmItM2VhODY4NjNmMGQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIxVDEzNDEwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEyOTcyMjJkYTY5ZDU3MDc1MWZlMjU1MTY2ZTUzZGUyMWNjN2FjMjRiMWRjODNjMzczZjBhYzgwNDViMzJmNWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BDHW_bFPcEY1puuqWMw40ssNPzfifVm05xPCipnDeDQ)

Are those small backup files maybe around 1kb? or lets say <100kb? 
I faced an issue a while ago, that even manually doing adb backups, the file was always around 1kb and nothing in it.
turned out, that sometimes there is an adb version missmatch between phone and the adb.exe on the computer.

So it is maybe worth it do google "how to get adb version on smartphone" and than download the according adb.exe and give it another try.

**Comment by adi-works** [comment]
Thanks. I matched the versions, but it still doesn't yield the important files. Although, the version on my phone has a weird email suffix. I tried matching it to this: https://dl.google.com/android/repository/platform-tools_r34.0.0-windows.zip.

![image](https://github.com/YuvrajRaghuvanshiS/WhatsApp-Key-Database-Extractor/assets/12032248/7d2d4067-3796-4709-9480-247c0a49a82d)

**Comment by betterthanyou155** [comment]
> I also noticed the script doesn't wait for the backup to end, seems like a funny racing condition

exact same problem here mate, in my case sometimes it creates a file of just 400 bytes, obv not the correct thing. and if i wait for a bit, it seems to catch on but the file size is still just 15MB, no key and msgstore.db present. Did you find any solution??

**Comment by Eco-Gaming** [comment]
I had the same problem for a while, I managed to fix it by using a non numeric password for the backup (`unicorn` in my case) and waiting around 30s before pressing the backup button on my phone. I also used the `--tar-only` argument, but I don't think that changed anything.

With that I exported ~311,000 messages today, with my Windows 10 PC and Galaxy S10, coming from the newest version of WhatsApp.

**Comment by anmac1789** [comment]
> I had the same problem for a while, I managed to fix it by using a non numeric password for the backup (`unicorn` in my case) and waiting around 30s before pressing the backup button on my phone. I also used the `--tar-only` argument, but I don't think that changed anything.
> 
> With that I exported ~311,000 messages today, with my Windows 10 PC and Galaxy S10, coming from the newest version of WhatsApp.

I am trying to extract from my note 8 on android 9 and I had to start the extractor by starting powershell and typing ./wa_kdbe.py --allow-reboot and then I was able to get the files but EVEN THEN there is an error produced that it can't get the key file

**Comment by tvulin** [comment]
I ran into this error as well.

I'm using v0.4.1 from https://pypi.org/project/wakdbe/ (I fixed an issue with the path) and it worked fine with 2 phones, I could merge databases and restore the result.

And then at some point (I did several extracts as I was debugging the merge script over several days when I had some time), I got this message (missing key) on one of the phones that worked previously. That was after restoring a msgstore with WhatsApp on it (worked fine, and I don't know if that's linked to the subsequent error).
